### PR TITLE
LogRow: Fixed background-height when hovering and label-alignment

### PR DESCRIPTION
**Pull request by svennergr**
**What this PR does / why we need it**:

While reviewing #51010 I noticed a small difference in heights in the LogRow. This PR fixes the height differences by adding a `position: relative` to every table-cell.

Additionally I noticed that log-lables are not vertically aligned inside a row. Fixed that by adjusting it to the right `margin-top`.

**Special notes for your reviewer**:

Before the fix:
image
Especially the height difference is hard to spot. I tried highlighting that by adding some blue lines.

After the fix:
Screenshot 2022-06-17 at 12 07 24

By the way, the reason for this difference is kinda weird. The `computed` tab for table-cells before the fix on cells already having the `position: relative` looks like this:
image

And for cells not having `position: relative` looks like this:
image

I went with the approach to add `position: relative` to every cell because it was just easier. At least the first cell indicating the log-level **needs** `position: relative`, so there wasn't really a way to remove it.

### CODE REVIEW

File: packages/grafana-ui/src/components/Logs/LogLabels.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => {
       padding: 0 2px;
       background-color: ${theme.colors.bg2};
       border-radius: ${theme.border.radius};
-      margin: 1px 4px 0 0;
+      margin: 1.75px 4px 0 0;
```
Comment (ivanahuckova) [inline packages/grafana-ui/src/components/Logs/LogLabels.tsx:26]: I am wondering if we need to change styling here, cause Log Labels is used at multiple places and it is going to miss-align this at other places. Can't we fix it directly in the cell?

Comment (ivanahuckova) [reply packages/grafana-ui/src/components/Logs/LogLabels.tsx:26]: It is for example used here 
image

Comment (svennergr) [reply packages/grafana-ui/src/components/Logs/LogLabels.tsx]: Good catch! I moved it to select the first `span` in the labels-cell instead. Do you think this is better?

Comment (ivanahuckova) [reply packages/grafana-ui/src/components/Logs/LogLabels.tsx]: Yeah! I think it make sense here.

**Review by ivanahuckova — APPROVED** [review]
LGTM

**Pull request merged**